feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go

```diff
@@ -324,13 +324,42 @@ func TestQoSCounters(t *testing.T) {
 
 func TestComponentParent(t *testing.T) {
 	dut := ondatra.DUT(t, "dut")
-	var componentParent = map[string]string{
-		"Fabric":      "Chassis",
-		"FabricChip":  "Fabric",
-		"Linecard":    "Chassis",
-		"PowerSupply": "Chassis",
-		"Supervisor":  "Chassis",
-		"SwitchChip":  "Linecard",
+	var componentParent = map[ondatra.Vendor]map[string]string{
+		ondatra.CISCO: {
+			"Fabric":      "Rack",
+			"FabricChip":  "Rack",
+			"Linecard":    "Rack",
+			"PowerSupply": "Rack",
+			"Supervisor":  "Rack",
+			"SwitchChip":  "0/",
+		},
+		ondatra.ARISTA: {
+			"Fabric":      "Chassis",
+			"FabricChip":  "Fabric",
+			"Linecard":    "Chassis",
+			"PowerSupply": "Chassis",
+			"Supervisor":  "Chassis",
+			"SwitchChip":  "Linecard",
+		},
+	}
```

Comment:
Should there be a default component list and a vendor specific list?  With this change we are restricting only to ARISTA and CISCO

Reply:
The code should be refactored to find components by the openconfig defined types.  For example, a NOS (network operating system) should use the appropriate type for their CONTROLLER_CARD and will use a NOS specific 'name' value for the. (routing-engine vs. supervisor, etc).

https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-platform.html#openconfig-platform-types-identities

Reply:
Example code to look up component name based on component identity (type):
https://github.com/openconfig/featureprofiles/blob/04860e1abb6139d8135a8f46fa46bb7c8553c128/internal/components/components.go#L40-L59

Reply:
@dplore , Changed to using FindComponentsByType